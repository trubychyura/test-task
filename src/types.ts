export type IComment = {
  text: string;
  rating: number;
};

export type IProduct = {
  asin: string;
  name: string;
  img: string;
  comments: IComment[];
  [key: string]: any;
};

export type HomeProps = {
  products: IProduct[];
  loadData: () => void;
};

export type RatingProps = {
  name?: string;
  stars?: number;
};

export type StarProps = {
  value: number;
  rating: number;
  hover?: number | null;
};

export type FormProps = {
  id: string;
  handleSubmit: (id: string, data: IComment) => void;
};

export type FormikValues = {
  comment: string;
  rating: number;
};

export type ProductProps = {
  id: string;
  name: string;
  img: string;
};

export type IState = {
  products: IProduct[];
};

export type ModalProps = {
  products: IProduct[];
};

export type ParamTypes = {
  id: string;
};
