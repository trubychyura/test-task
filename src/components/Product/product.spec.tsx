import { shallow, ShallowWrapper } from 'enzyme';
import { Link } from 'react-router-dom';

import { Product } from '..';
import { ProductProps } from '../../types';

const setUp = (props: ProductProps): ShallowWrapper =>
  shallow(<Product {...props} />);

describe('Product component', () => {
  let component: ShallowWrapper;
  let mockProps: ProductProps;
  beforeEach(() => {
    mockProps = {
      id: 'mock id',
      name: 'mock name',
      img: 'mock img',
    };
    component = setUp(mockProps);
  });

  it('Link should contain proper text', () => {
    const wrapper = component.find(Link);
    expect(wrapper.text()).toEqual(mockProps.name);
  });

  it('should render Product component', () => {
    expect(component).toMatchSnapshot();
  });
});
