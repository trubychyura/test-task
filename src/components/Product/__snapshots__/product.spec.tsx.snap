// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should render Product component 1`] = `
<Styled(Row)>
  <Styled(Col)
    md={
      Object {
        "order": "first",
      }
    }
    xs={
      Object {
        "order": "last",
        "span": 12,
      }
    }
  >
    <Link
      to="/mock id"
    >
      mock name
    </Link>
  </Styled(Col)>
  <Styled(Col)
    md={
      Object {
        "order": "last",
      }
    }
    xs={
      Object {
        "order": "first",
        "span": 12,
      }
    }
  >
    <Styled(Image)
      src="mock img"
    />
  </Styled(Col)>
</Styled(Row)>
`;
